src/schemathesis/models.py

```diff
@@ -176,10 +177,12 @@ def call(
             close_session = True
         else:
             close_session = False
-
         data = self.as_requests_kwargs(base_url)
+        data["headers"] = data["headers"] or {}
         if headers is not None:
-            data["headers"] = {**(data["headers"] or {}), **headers}
+            data["headers"].update(headers)
```

Comment:
There is a problem with updating this key - it references the original `headers` dictionary (if it is not None) that is in the `self.headers` attribute. So if you are updating it, then it mutates that dictionary - if it is used in some other places in the test after `case.call`, then it might not be desired to have it mutated. I suggest adding `data["headers"] = data["headers"].copy()` few lines above (after this line - `data["headers"] = data["headers"] or {}`)

In the future, I plan to refactor it and make `as_requests_kwargs` and `as_werkzeug_kwargs` return copies to avoid unintentional mutations

Reply:
Ok